lib/ecl/ecl_smspec.cpp

```diff
@@ -1287,18 +1287,28 @@ int ecl_smspec_get_num_regions(const ecl_smspec_type * ecl_smspec) {
 
 /*****************************************************************/
 
-#define NODE_RETURN_INDEX(node_ptr)                     \
-  if (!node_ptr)                                        \
-    return -1;                                          \
-  else                                                \
-    return smspec_node_get_params_index( node_ptr );
 
 
-#define NODE_RETURN_EXISTS(node_ptr)            \
-  if (!node_ptr)                                \
-    return false;                               \
-  else                                          \
-    return true;
+
+namespace {
+
+  bool NODE_EXISTS(const ecl::smspec_node * node_ptr) {
+    if (node_ptr)
+      return true;
+
+    return false;
+  }
+
+
+  int NODE_VALID_INDEX(const ecl::smspec_node * node_ptr) {
+    if (node_ptr)
+      return node_ptr->get_params_index();
+
+    throw std::invalid_argument("Invalid lookup summary object");
```

Comment:
I think an `out_of_bounds` exception would be more in line with STL behavior at least, i.e. `map::at`.

Reply:
I have updated to `std::out_of_range()` - feel the `std::` exception hierarchy is slightly limited - and nearly always end up on `std::invalid_argument` - but yes; I agree `std::out_of_range()` is a better choice.